eng/ApiCompat/ApiCompat.targets

```diff
@@ -0,0 +1,131 @@
+      <_WindowsFormsApiCompatBaselineIgnoreFile>$(MSBuildThisFileDirectory)baseline.net$(_BaselineMajorMinorVersion).winforms.txt</_WindowsFormsApiCompatBaselineIgnoreFile>
+      <_WpfApiCompatBaselineIgnoreFile>$(MSBuildThisFileDirectory)baseline.net$(_BaselineMajorMinorVersion).wpf.txt</_WpfApiCompatBaselineIgnoreFile>
+    </PropertyGroup>
+
+    <Error Text="Windows Forms ref assemblies cannot be found. Expected under '$(_WindowsFormsRefsPath)'."
+           Condition="!Exists('$(_WindowsFormsRefsPath)')"/>
+    <Error Text="Windows Forms API compat baseline for net$(_BaselineMajorMinorVersion) must be created and stored in '$(_WindowsFormsApiCompatBaselineIgnoreFile)'."
+           Condition="!Exists('$(_WindowsFormsApiCompatBaselineIgnoreFile)') and '$(CreateBaseline)' != 'true'"/>
+    <!-- TODO: WPF team to enable and review
+       <Error Text="WPF API compat baseline for net$(_BaselineMajorMinorVersion) must be created and stored in '$(_WpfApiCompatBaselineIgnoreFile)'."
+           Condition="!Exists('$(_WpfApiCompatBaselineIgnoreFile)') and '$(CreateBaseline)' != 'true'"/> -->
+
+    <ItemGroup>
+      <_DependencyDirectories Include="$(_NETCoreAppRefPath)" />
+      <_DependencyDirectories Include="$(_WindowsDesktopRefPath)" />
+      <_ContractDependencyDirectories Include="$(_NETCoreAppRuntimePath)" />
+
+      <!-- Get the list of Windows Forms ref assemblies shipped in the transport package -->
+      <_WindowsFormsFilePaths Include="$(_WindowsFormsRefsPath)%(FrameworkListFileClass.Identity)"
+                              Condition="'%(FrameworkListFileClass.Profile)' == 'WindowsForms'" />
+      <_WindowsFormsFilePaths Remove="@(_WindowsFormsFilePaths)"
+                              Condition="!Exists(%(Identity))" />
+
+      <!-- Get the list of WPF assemblies that form the Ref surface, i.e. go into <dotnet>\packs\Microsoft.WindowsDesktop.App.Ref\<version>\ref folder -->
+      <_WpfFilePaths Include="$(OutputPath)%(FrameworkListFileClass.Identity)"
+                              Condition="'%(FrameworkListFileClass.Profile)' == 'WPF'" />
+    </ItemGroup>
+
+    <!-- Validate paths for the collected items -->
+    <Error Text="One ofthe required contract dependencies directories is invalid."
+           Condition="!Exists('%(_ContractDependencyDirectories.FullPath)')"/>
+    <Error Text="One ofthe required implementation directories is invalid."
+           Condition="!Exists('%(_DependencyDirectories.FullPath)')"/>
+    <Error Text="Some of Windows Forms assemblies referenced in 'FrameworkListFileClass' group can't be found."
+           Condition="!Exists('%(_WindowsFormsFilePaths.FullPath)')"/>
+
+    <PropertyGroup>
+      <_WindowsFormsApiCompatArgs>"@(_WindowsFormsFilePaths, ',')"</_WindowsFormsApiCompatArgs>
+      <_WindowsFormsApiCompatArgs>$(_WindowsFormsApiCompatArgs) --contract-depends "$(OutputPath),@(_ContractDependencyDirectories, ','),"</_WindowsFormsApiCompatArgs>
```

Comment:
My understanding is that the purpose of this comparison in servicing is that you're trying to make sure that you don't expose any new surface area over a baseline.  

In other words, you want to make sure the `contract` established by your `current` build is satisfied by the `implementation` of your `baseline`.

I believe you have the right data flowing in, but it helps to define these terms and make it crystal clear what the comparison is doing.  I also recommend using `--left-operand current --right-operand baseline` so that the error messages are clearer.